tests/context_storages/test_dbs.py

```diff
@@ -148,9 +148,9 @@ def test_redis(testing_context, context_id):
 def test_postgres(testing_context, context_id):
     db = context_storage_factory(
         "postgresql+asyncpg://{}:{}@localhost:5432/{}".format(
-            os.getenv("POSTGRES_USERNAME"),
-            os.getenv("POSTGRES_PASSWORD"),
-            os.getenv("POSTGRES_DB"),
+            os.environ["POSTGRES_USERNAME"],
+            os.environ["POSTGRES_PASSWORD"],
+            os.environ["POSTGRES_DB"],
```

Comment:
I think db tests should skip if env vars are not set, not fail.

Reply:
I think there should be some explicit parameters (environment variables?) for enabling/disabling DB testing.
Because right now the reasons for skipping tests do not seem to be obvious: the tests will be skipped if the dependencies are not available or when environment variables are not set - but that can also be result of some kind of configuration failure and in that case I guess tests should be failed.

Reply:
There is a way to disable certain tests from running:
https://docs.pytest.org/en/latest/example/markers.html#mark-examples

The way I see it:
A test fails if something went wrong that the test was supposed to test (e.g. context storage returned a different context)
A test skips if something went wrong that the test was not supposed to test (e.g. database is not ready for testing)

This approach allows someone without, for example, telegram bot token to run tests and get a success.

This is confirmed by the [documentation for pytest](https://docs.pytest.org/en/latest/how-to/skipping.html):
> A skip means that you expect your test to pass only if some conditions are met, otherwise pytest should skip running the test altogether. Common examples are skipping windows-only tests on non-windows platforms, or skipping tests that depend on an external resource which is not available at the moment (for example a database).

But I agree with you that there is a problem:

Sometimes we expect a test to not skip but if it skips our check is still considered successful.
Of the top of my head I see one solution: we create an env variable that stores tests that we expect to skip. If a test is not in that list, it cannot be skipped.
I will try to think of other solutions to the problem.

------------

Also, another thought:
Maybe we shouldn't use env in tests at all.
And, in this case, pass a hardcoded URI to the factory.

Downside:
Have to change the hardcoded URI whenever we change env vars (something that is not of a big concern, we never change them)
Upside:
test become `.env_file` independent

Reply:
I would prefer the first solution, because that way tests also test `.env_file` integration itself.